elevate-project/controllers/project.js

```diff
@@ -1,29 +1,75 @@
+const { transform } = require('lodash')
 const routeConfigs = require('../constants/routes')
 const requesters = require('../utils/requester')
 const requestParser = require('../utils/requestParser')
 
-// const createUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body)
-// }
-// const updateUser = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-
-// 	//const filteredRequestBody = requestParser.transformUpdateUserBody(req.body)
-// 	return await requesters.patch(req.baseUrl, selectedConfig.targetRoute.path, req.body, req.headers)
-// }
-
-// const entityTypeRead = async (req, res, responses) => {
-// 	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
-// 	return await requesters.post(req.baseUrl, selectedConfig.targetRoute.path, req.body, {
-// 		'X-auth-token': req.headers['x-auth-token'],
-// 	})
-// }
-
-const userController = {
-	// createUser,
-	// updateUser,
-	// entityTypeRead,
+const fetchProjectTemplates = async (req, res, responses) => {
+	const selectedConfig = routeConfigs.routes.find((obj) => obj.sourceRoute === req.sourceRoute)
+
+	let response = {result: {data: [],count: 0}}
+	let proceed = true
+
+	// fetch the max limit from the env file for the DB Find API
+	const max_limit = process.env.RESOURCE_MAX_FETCH_LIMIT ? parseInt(process.env.RESOURCE_MAX_FETCH_LIMIT, 10) : 1000
+	// construct body for fetching project templates 
+	let body = {
+		"query": {
+			"status": "published"
+		},
+		"projection": ["_id" , "title","createdBy","createdAt","description"],
+		"limit" : max_limit
+	}
+	// custom header 
+	let header = {
+		'internal-access-token' : req.headers['internal_access_token'],
+		'Content-Type' : 'application/json'
+	}
+	// enrich body if needed 
+	if(req.body){
+		if(req.body.search) body.query.title = {
+			"$regex" : req.body.search,
+			"$options" : 'i'
+		}
+
+		// check if any specific resource type is requested 
+		if(req?.body?.resourceType !== undefined || req?.body?.resourceType != []){
+			// flag to abort the api call if projects is not requested 
+			proceed = (req?.body?.resourceType?.length != 0 && req?.body?.resourceType.includes('projects')) ? true : (req?.body?.resourceType?.length == 0 ? true : false)
```

Comment:
@adithyadinesh0412 this we can check first of the resourceType is not there then we dont need to construct the query and projection and all

Reply:
We need to construct either way. bec if resource type is not there, we need to pass all the resources. So we will need query.
From frontend if there is no type filter they will be giving type empty. @priyanka-TL